Purpose: Fix kernel panic on kfreebsd
Fixes:   #672959
Author:  Petr Salinger
Status:  Not yet submitted upstream
---
--- a/startpar/startpar.c	2012-08-21 22:02:07.000000000 +0100
+++ b/startpar/startpar.c	2012-08-21 22:02:41.340814312 +0100
@@ -1122,9 +1122,11 @@
     }
 #endif
 
+#ifdef __linux__
   /* lock us into memory */
   if (geteuid() == 0)
     mlockall(MCL_CURRENT|MCL_FUTURE);
+#endif
   errno = 0;
 
   gettimeofday(&glastio, 0);
